### PR TITLE
Homework, mostly complete.

### PR DESCRIPTION
**Pull request by edanweis**
Had a bit of trouble converting entries view and forms to ModelForms and Class Based Views.

### CODE REVIEW

File: timetracker/entries/forms.py [hunk 1]
```diff
@@ -4,14 +4,17 @@
...
-	name = forms.CharField()
-	client = forms.ModelChoiceField(queryset=Client.objects.all())
-
+class ProjectForm(forms.ModelForm):
+    class Meta:
+        client = forms.ModelChoiceField(queryset=Client.objects.all())
+        model = Project
+        fields =['name', 'client']
 
 class EntryForm(forms.Form):
```
Comment (MarkusH) [inline timetracker/entries/forms.py:19]: Any reason you didn't convert this form to use a `ModelForm`?

File: timetracker/entries/urls.py [hunk 2]
```diff
@@ -2,11 +2,14 @@
 
 from . import views
 
+from .models import Client
+from .views import ClientCreateView, ProjectCreateView, RootRedirectView
 
 urlpatterns = [
-    url(r'^clients/$', views.clients, name='client-list'),
+	url(r'^$', RootRedirectView.as_view(), name='root'),
```
Comment (MarkusH) [inline timetracker/entries/urls.py:9]: You should be careful with using tabs and spaces for indentation. Recommendation is to use 4 spaces.

File: timetracker/entries/views.py [hunk 3]
```diff
@@ -1,30 +1,25 @@
...
-        form = ClientForm()
+class ClientCreateView(CreateView):
+    """
+    View to create a Client instance every
+    time the form is submitted
+    """
+    model = Client
+    template_name = 'clients.html'
+    form_class = ClientForm
+    success_url = '/clients/'
```
Comment (MarkusH) [inline timetracker/entries/views.py:17]: You can use `django.core.urlresolvers.reverse_lazy()` to not write hard-coded paths

File: timetracker/entries/views.py [hunk 4]
```diff
@@ -68,23 +63,22 @@ def entries(request):
...
-                client=form.cleaned_data['client']
-            )
-    else:
-        form = ProjectForm()
+class ProjectCreateView(CreateView):
+    """
+    View to create a Project instance every
+    time the form is submitted
+    """
+    project = Project.objects.get(pk=1)
```
Comment (MarkusH) [inline timetracker/entries/views.py:71]: I don't think you want to have this project here.

File: timetracker/entries/views.py [hunk 5]
```diff
@@ -68,23 +63,22 @@ def entries(request):
...
+    model = Project
+    template_name = 'projects.html'
+    form_class = ProjectForm
+    success_url = '/projects/'
+    initial = {'name':project.name}
```
Comment (MarkusH) [inline timetracker/entries/views.py:76]: ... neither this line, as they are executed the first time this python file is loaded and won't change afterwards.

File: timetracker/entries/forms.py (hunk 1 above)
Comment (edanweis) [inline timetracker/entries/forms.py:19]: I think I had some difficulty with this one. Can't recall exactly why

File: timetracker/entries/views.py (hunk 5 above)
Comment (edanweis) [inline timetracker/entries/views.py:76]: Yes, sorry I think I left those lines in there by mistake

File: timetracker/entries/views.py [hunk 6]
```diff
@@ -106,3 +100,24 @@ def project_detail(request, pk):
...
+
+    def get_context_data(self, **kwargs):
+        context = super(ProjectCreateView, self).get_context_data(**kwargs)
+        context['projects'] = self.model.objects.all()
+        return context
+
+class RootRedirectView(RedirectView):
+    permanent = False
+
+    def get_redirect_url(self):
```
Comment (MarkusH) [inline timetracker/entries/views.py:122]: You can use `redirect_url = reverse_lazy('client-list')` instead :wink: (`reverse_lazy` is at the same place as `reverse`)

**Comment by MarkusH** [comment]
Looks good :smile:

**Pull request closed without merging**